alacritty_terminal/src/term/mod.rs

```diff
@@ -927,13 +936,19 @@ impl<T> Term<T> {
         self.damage.damage_point(point);
     }
 
+    #[inline]
+    fn damage_grid_point(&mut self, point: Point) {
+        let display_offset = self.grid.display_offset() as i32;
+        let lines = self.screen_lines() as i32 - 1;
+        let line = (display_offset + point.line.0).clamp(0, lines) as usize;
+
+        self.damage.damage_point(Point::new(line, point.column));
+    }
+
     /// Damage `Vi` mode cursor.
     #[inline]
     pub fn damage_vi_cursor(&mut self) {
-        let line = (self.grid.display_offset() as i32 + self.vi_mode_cursor.point.line.0)
-            .clamp(0, self.screen_lines() as i32 - 1) as usize;
-        let vi_point = Point::new(line, self.vi_mode_cursor.point.column);
-        self.damage.damage_point(vi_point);
+        self.damage_grid_point(self.vi_mode_cursor.point)
```

Comment:
Might not be worth it to have a method to do so at this point. Seems like that would be pretty concise just inlined.